android/KMEA/app/src/main/assets/keyboard.html

```diff
@@ -130,7 +130,8 @@
 
     function setKeymanLanguage(k) {
       KeymanWeb.registerStub(k);
-      keyman.setActiveKeyboard(k.KP + '::'+k.KI, k.KLC);
+      // KMW will handle the k.KP:: namespacing
+      keyman.setActiveKeyboard(k.KI, k.KLC);
```

Comment:
While KMW adds namespacing when the stub is registered... I do not see in the code where it does the same when setting the active keyboard.  Or even where `setActiveKeyboard` de-namespaces a stub's ID to attempt a match.

Could you please point me to the code block that handles this case?  I'd expect to see something within `_SetActiveKeyboard` ([link](https://github.com/keymanapp/keyman/blob/754430eff7e84ffa42c065bbba14bb7b170a42c0/web/source/keyboards/kmwkeyboards.ts#L452)).

Reply:
https://github.com/keymanapp/keyman/blob/651932f1bbbb6d7f724b2e602e0c042f227bd982/web/source/kmwembedded.ts#L165

It was confusing me why the call became
`setActiveKeyboard(sil_euro_latin::sil_euro_latin::Keyboard_sil_euro_latin, en)`

Reply:
Oh... we're getting a pass-by-reference effect.  We should probably add a comment to indicate that `KeymanWeb.registerStub(k)` will mutate `k.KI`, but that said mutation matches what `setActiveKeyboard` will be looking for.

Reply:
So... how was this not constantly breaking things before?

Reply:
It's a mystery to me 😞. But was pretty recent

Reply:
It might be worth expending the effort to find out how this broke.

Reply:
I understand how this broke. It was due to the refactor in #5338 which did not anticipate registerStub mutating its parameter. Perhaps we can fix registerStub and then leave this as is?

Reply:
The easiest such "fix" would be to make a shallow clone of the `registerStub`'s parameter and using the cloned instance for registration.  Nothing too complex; just a bit tedious.

Reply:
Can @jahorton update registerStub?

I note mutating Pstub seems intentional as follows
https://github.com/keymanapp/keyman/blob/75e47c1ccc36e1698bafa8178acf8140f88e4e2b/web/source/keyboards/kmwkeyboards.ts#L1539-L1548

Reply:
Mutating the version _inside the method_ yes.  _Outside_ the method (pass-by-reference), not so much.  Just make a shallow clone and mutate that one.

Reply:
Just add this special sauce at the start of registerStub 🍅🧴:
```js
Pstub = { ... Pstub };  // shallow clone the stub object
```